cores/esp32/esp32-hal-adc.c

```diff
@@ -142,9 +144,9 @@ bool __adcAttachPin(uint8_t pin){
 #endif
     }
 #if CONFIG_IDF_TARGET_ESP32 || CONFIG_IDF_TARGET_ESP32S2
```

Comment:
while here, let's use the SOC defines te test wether DAC is supported and maybe how many channels

Reply:
SOC define used now.
According to how many channels there are always 2 channels when supported.

Reply:
It's not given that this will be true in the future. Why not have more?